### PR TITLE
Test slurm commands worker in CI

### PR DESCRIPTION
**Pull request by jtronge**
This should add testing for the slurm commands version of the worker in CI, addressing #823.

I'm labeling this WIP until the tests pass.

### CODE REVIEW

**Comment by rstyd** [comment]
This looks great! I think we should hold off on merging this until the restart GDB pr is merged just cause I don't want to have to rebase it again. lol

**Review by rstyd — APPROVED** [review]

**Comment by pagrubel** [comment]
Ugh this is the one I'd like to understand better before merging. I guess we should just hold off on merging anything right now.

**Pull request merged**
